handsontable/src/3rdparty/walkontable/src/utils/stylesHandler.js

```diff
@@ -0,0 +1,188 @@
+      return this.#cssVars[`--ht-${variableName}`];
+    }
+
+    const acquiredValue = this.#getParsedCSSValue(`--ht-${variableName}`);
+
+    if (acquiredValue && acquiredValue !== '') {
+      this.#cssVars[variableName] = acquiredValue;
+
+      return acquiredValue;
+    }
+  }
+
+  /**
+   * Retrieves the computed style value for a specified CSS property of a `td` element.
+   *
+   * @param {string} cssProperty - The CSS property to retrieve the value for.
+   * @returns {number|string|undefined} The value of the specified CSS property, or `undefined` if not found.
+   */
+  getStyleForTD(cssProperty) {
+    return this.#computedStyles?.td[cssProperty];
+  }
+
+  /**
+   * Calculates the row height based on the current theme and CSS variables.
+   *
+   * @returns {number} The calculated row height.
+   */
+  getDefaultRowHeight() {
+    if (this.#isClassicTheme) {
+      return CLASSIC_THEME_DEFAULT_HEIGHT;
+    }
+
+    return this.getStyleForTD('border-top-width') +
+      (this.getCSSVariableValue('cell-vertical-padding') * 2) +
+      this.getCSSVariableValue('line-height') +
+      this.getStyleForTD('border-bottom-width');
+  }
+
+  /**
+   * Retrieves and processes the computed styles for a `td` element.
+   *
+   * This method creates a temporary table structure, appends it to the root element,
+   * retrieves the computed styles for the `td` element, and then removes the table
+   * from the DOM. The computed styles are passed to the provided callback function.
+   *
+   * @param {Array} cssProps - An array of CSS properties to retrieve.
+   * @returns {object} An object containing the requested computed styles for the `td` element.
+   * @private
+   */
+  #getStylesForTD(cssProps) {
+    const rootDocument = this.#rootDocument;
+    const rootElement = this.#rootElement;
+    const table = rootDocument.createElement('table');
+    const tbody = rootDocument.createElement('tbody');
+    const tr = rootDocument.createElement('tr');
+    // This needs not to be the first row in order to get "regular" vaules.
+    const tr2 = rootDocument.createElement('tr');
+    const td = rootDocument.createElement('td');
+
+    tr2.appendChild(td);
+    tbody.appendChild(tr);
+    tbody.appendChild(tr2);
+    table.appendChild(tbody);
+
+    rootElement.appendChild(table);
+
+    const computedStyle = getComputedStyle(td);
+    const returnObject = {};
+
+    cssProps.forEach((prop) => {
+      returnObject[prop] = computedStyle.getPropertyValue(prop);
+    });
+
+    rootElement.removeChild(table);
+
+    return returnObject;
+  }
+
+  /**
+   * Parses the value of a specified CSS property from the root element's computed style.
+   *
+   * @param {string} property - The CSS property to retrieve and parse.
+   * @returns {number|string} The parsed value of the CSS property.
+   */
+  #getParsedCSSValue(property) {
+    return parseInt(this.#rootComputedStyle?.getPropertyValue(property), 10);
+  }
```

Comment:
The `getPropertyValue` method returns an empty string for non-existing property, which, after casting to a number, returns `NaN.` WDYT about improving the method?
```suggestion
  /**
   * Parses the value of a specified CSS property from the root element's computed style.
   *
   * @param {string} property - The CSS property to retrieve and parse.
   * @returns {number|undefined} The parsed value of the CSS property.
   */
  #getParsedCSSValue(property) {
    const parsedValue = parseInt(this.#rootComputedStyle.getPropertyValue(property), 10);
    
    return Number.isNaN(parsedValue) ? null : parsedValue;
  }
```
then it is worth updating the `getCSSVariableValue` method with the changes made here.